llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8860,12 +8875,16 @@ VPRecipeBuilder::getScaledReduction(PHINode *PHI,
   Instruction *ExtA = cast<Instruction>(BinOp->getOperand(0));
   Instruction *ExtB = cast<Instruction>(BinOp->getOperand(1));
 
+  // Check that the extends extend from the same type.
+  if (A->getType() != B->getType())
+    return std::nullopt;
+
```

Comment:
The target checks this.

Reply:
It used to be checked here until the second re-land of the partial reduction work so it could be back here because of rebasing gone wrong!